clients/include/testing_csx2dense.hpp

```diff
@@ -0,0 +1,396 @@
+    {
+        PRINT_IF_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    //
+    // Testing invalid handle.
+    //
+    status = csx2dense(nullptr, 0, 0, nullptr, (const T*)nullptr, nullptr, nullptr, (T*)nullptr, 0);
+    verify_hipsparse_status_invalid_handle(status);
+
+    //
+    // Testing invalid pointers.
+    //
+    status = csx2dense(handle,
+                       M,
+                       N,
+                       nullptr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(handle,
+                       M,
+                       N,
+                       descr,
+                       (const T*)nullptr,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(
+        handle, M, N, descr, (const T*)d_csx_val, nullptr, d_csx_col_row_ind, (T*)d_dense_val, LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(
+        handle, M, N, descr, (const T*)d_csx_val, d_csx_row_col_ptr, nullptr, (T*)d_dense_val, LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    status = csx2dense(handle,
+                       M,
+                       N,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)nullptr,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid pointer must be detected.");
+
+    //
+    // Testing invalid size on M
+    //
+    status = csx2dense(handle,
+                       -1,
+                       N,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid size must be detected.");
+    //
+    // Testing invalid size on N
+    //
+    status = csx2dense(handle,
+                       M,
+                       -1,
+                       descr,
+                       (const T*)d_csx_val,
+                       d_csx_row_col_ptr,
+                       d_csx_col_row_ind,
+                       (T*)d_dense_val,
+                       LD);
+    verify_hipsparse_status_invalid_pointer(status, "Error: an invalid size must be detected.");
```

Comment:
Should be verify_hipsparse_status_invalid_size?

Reply:
I guess it works here because the pointer one just defaults to invalid value internally but given that we have the verify_hipsparse_status_invalid_size method probably we should use it here?

Reply:
Yes :) fixed.